addons/udes_warehouse_classification/tests/test_product_template.py

```diff
@@ -0,0 +1,59 @@
+from .common import Base
+
+
+class TestWarehouseClassificationPicking(Base):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.test_messages_01 = ["Inbound, Report a, Report"]
+        cls.test_messages_02 = ["Report a Report", "Report b Report"]
+        cls.test_messages_03 = ["Report a Report", "Inbound, Report a, Report"]
+
+    # functions to test:  and _get_classification_messages_for_product_picking
+
+    # For a given product find the classification messages for specified report,
+    # 1. report a on it's own - banana
+    # 2. not matching report type - banana
+    # 3. report a and b - honeydew
+    # 4. report a and report a on inbound (i.e. 2 messages for a report) - cherry
+    # 5. no classifications - apple
+
+    def test01_basic_report(self):
+        messages = self.banana.product_tmpl_id.get_classification_messages_for_report(
+            self.report_a.report_name
+        )
+        self.assertEqual(messages, self.test_messages_01)
+
+    def test02_report_not_for_product(self):
+        self.assertFalse(
+            self.banana.product_tmpl_id.get_classification_messages_for_report(
+                self.report_b.report_name
+            )
+        )
+
+    def test03_multiple_reports(self):
+        messages_1 = self.honeydew.product_tmpl_id.get_classification_messages_for_report(
+            self.report_a.report_name
+        )
+        messages_2 = self.honeydew.product_tmpl_id.get_classification_messages_for_report(
+            self.report_b.report_name
+        )
+        self.assertIn(messages_1[0], self.test_messages_02)
+        self.assertIn(messages_2[0], self.test_messages_02)
+
+    def test04_2_messages_one_product(self):
+        messages = self.cherry.product_tmpl_id.get_classification_messages_for_report(
+            self.report_a.report_name
+        )
+        self.assertCountEqual(messages, self.test_messages_03)
+
+    def test05_no_classfications_for_product(self):
+        with self.assertLogs(
+            "odoo.addons.udes_warehouse_classification.models.product_template"
+        ) as cm:
+            messages = self.apple.product_tmpl_id.get_classification_messages_for_report(
+                self.report_b.report_name
+            )
+        self.assertFalse(messages)
+        self.assertIn("Product Test product Apple has no warehouse classifications.", cm.output[0])
```

Comment:
What about multiple products? It is api.multi after all

Reply:
fixed